Candidate: CVE-2024-26130
PublicDate: 2024-02-22
References:
 https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-26130
 https://github.com/pyca/cryptography/security/advisories/GHSA-6vqw-3v5j-54x4
 https://github.com/pyca/cryptography/pull/10423
 https://github.com/pyca/cryptography/commit/97d231672763cdb5959a3b191e692a362f1b9e55
Description:
 cryptography is a package designed to expose cryptographic primitives and
 recipes to Python developers. Starting in version 38.0.0 and prior to
 version 42.0.4, if `pkcs12.serialize_key_and_certificates` is called with
 both a certificate whose public key did not match the provided private key
 and an `encryption_algorithm` with `hmac_hash` set (via
 `PrivateFormat.PKCS12.encryption_builder().hmac_hash(...)`, then a NULL
 pointer dereference would occur, crashing the Python process. This has been
 resolved in version 42.0.4, the first version in which a `ValueError` is
 properly raised.
Ubuntu-Description:
Notes:
Mitigation:
Bugs:
Priority: medium
Discovered-by:
Assigned-to:
CVSS:

Patches_python-cryptography:
upstream_python-cryptography: needs-triage
trusty_python-cryptography: DNE
xenial_python-cryptography: ignored (end of standard support)
esm-infra/xenial_python-cryptography: needs-triage
bionic_python-cryptography: ignored (end of standard support)
esm-infra/bionic_python-cryptography: needs-triage
focal_python-cryptography: needs-triage
jammy_python-cryptography: needs-triage
mantic_python-cryptography: needs-triage
devel_python-cryptography: needs-triage
